waspc/e2e-test/Tests/WaspCompileTest.hs

```diff
@@ -0,0 +1,20 @@
+module Tests.WaspCompileTest (waspCompile) where
+
+import GoldenTest (GoldenTest, runGoldenTest)
+import ShellCommands
+  ( cdIntoCurrentProject,
+    combineMakeShellCommands,
+    waspCliCompile,
+    waspCliNew,
+  )
+
+waspCompile :: GoldenTest
+waspCompile = do
+  let makeShellCommand =
+        combineMakeShellCommands
+          [ waspCliNew,
+            cdIntoCurrentProject,
+            waspCliCompile
+          ]
+
+  runGoldenTest "waspCompile" makeShellCommand
```

Comment:
A bit interesting that something named `GoldenTest` is calling `runGoldenTest` inside it. Is it really a golden test then? If it is running a golden test? What is it then? Not sure hm. Maybe then it is just `IO TestTree`? Did something change, did we name something else `GoldenTest` before? Or was it just `IO TestTree` all the time?
Maybe we are not running any golden test, but we are running golden test that is based on shell commands? So `runShellCmdGoldenTest`?
What is a `GoldenTest` for us really? How can it be `IO TestTree`, what does that mean?
What if `Main.hs` would have `runGoldenTest` mapped on all these tests? Or maybe those names are assigned there instead of here -> you could name the tests there!
A bunch of questions / ideas.

Maybe it is fine, maybe I am just overthinking this!

Reply:
We did have it taking a param I think, but now it is just an `IO TestTree` again, so let's go back to that and not hide it. Good call 👍🏻 I can also see what mapping run over the tests looks like too.